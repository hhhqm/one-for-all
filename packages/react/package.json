{
  "name": "@ofa/react",
  "version": "17.0.2",
  "main": "index.js",
  "license": "Apache-2.0",
  "scripts": {
    "build": "echo building react 17.0.2 done!"
  }
}
