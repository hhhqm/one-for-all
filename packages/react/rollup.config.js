import commonjs from "@rollup/plugin-commonjs";
import resolve from "@rollup/plugin-node-resolve";
import replace from "@rollup/plugin-replace";
import { terser } from "rollup-plugin-terser";

import packageJSON from './package.json';

export default [
  {
    input: `src/development.js`,
    output: {
      format: 'system',
      file: `dist/react.${packageJSON.version}.js`,
      sourcemap: 'inline',
      exports: "named",
    },
    plugins: [
      commonjs(),
      resolve(),
      replace({
        values: {
          "process.env.NODE_ENV": '"development"',
        },
      }),
    ],
  },
  {
    input: `src/production.js`,
    output: {
      format: 'system',
      file: `dist/react.${packageJSON.version}.min.js`,
      exports: "named",
    },
    plugins: [
      commonjs(),
      resolve(),
      replace({
        values: {
          "process.env.NODE_ENV": '"production"',
        },
      }),
      terser({
        output: {
          comments(node, comment) {
            return comment.value.trim().startsWith("react@");
          },
        },
      }),
    ],
  }
];
