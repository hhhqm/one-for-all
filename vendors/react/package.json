{
  "name": "@one-for-all/react",
  "version": "17.0.1",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "bundle": "rm -rf dist & rollup -c",
    "build": ""
  },
  "devDependencies": {
    "@rollup/plugin-commonjs": "^21.0.1",
    "react": "^17.0.2"
  }
}
