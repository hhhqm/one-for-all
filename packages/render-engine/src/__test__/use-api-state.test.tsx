import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import mockXHR from 'xhr-mock';

import petStoreSpec from '@ofa/spec-interpreter/src/__test__/petstore-spec';
import type { APIInvokeProperty, APIDerivedProperty } from '../types';
import StateHub from '../state-hub';
import Link from './link';

beforeEach(() => mockXHR.setup());
afterEach(() => mockXHR.teardown());

const stateIDMap = { stream_findPetsByTags: 'findPetsByTags' };
const stateHub = new StateHub(petStoreSpec, stateIDMap);

test('Link_changes_the_class_when_hovered', async () => {
  const mockRes = { data: { id: 'abc-123' } };
  mockXHR.get(/.*/, (req, res) => res.status(200).body(JSON.stringify(mockRes)));

  const onSuccessFn = jest.fn();

  const props: Record<string, APIDerivedProperty | APIInvokeProperty> = {
    foo: {
      type: 'api_derived_property',
      initialValue: 'foo',
      stateID: 'stream_findPetsByTags',
      convertor: () => 'abc',
    },
    bar: {
      type: 'api_derived_property',
      initialValue: 'bar',
      stateID: 'stream_findPetsByTags',
      convertor: () => 'abc',
    },
    onFetch: {
      type: 'api_invoke_property',
      stateID: 'stream_findPetsByTags',
      convertor: () => undefined,
      onSuccess: onSuccessFn,
    },
  };
  const { container, getByText } = render(<Link props={props} stateHub={stateHub} />);
  await waitFor(() => expect(onSuccessFn).toBeCalledTimes(1));
  await waitFor(() => getByText('abc:abc'));
  expect(container).toMatchSnapshot();
});

test('search_btn', async () => {
  const mockRes = { data: { id: 'abc-123' } };
  mockXHR.get(/.*/, (req, res) => res.status(200).body(JSON.stringify(mockRes)));

  const onSuccessFn = jest.fn();

  const props: Record<string, APIDerivedProperty | APIInvokeProperty> = {
    foo: {
      type: 'api_derived_property',
      initialValue: 'foo',
      stateID: 'stream_findPetsByTags',
      convertor: () => 'abc',
    },
    bar: {
      type: 'api_derived_property',
      initialValue: 'bar',
      stateID: 'stream_findPetsByTags',
      convertor: () => 'abc',
    },
    onFetch: {
      type: 'api_invoke_property',
      stateID: 'stream_findPetsByTags',
      convertor: () => undefined,
      onSuccess: onSuccessFn,
    },
  };
  const { container, getByText } = render(<Link props={props} stateHub={stateHub} />);
  await waitFor(() => expect(onSuccessFn).toBeCalledTimes(1));
  fireEvent(getByText('abc:abc'), new Event('click'));
  await waitFor(() => getByText('abc:abc'));
  expect(container).toMatchSnapshot();
});

