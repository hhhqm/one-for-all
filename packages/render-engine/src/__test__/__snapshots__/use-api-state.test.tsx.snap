// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link_changes_the_class_when_hovered 1`] = `
<div>
  <button
    id="button"
  >
    abc:abc
  </button>
</div>
`;
